src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -132,11 +132,8 @@ private static IEnumerable<TSource> TakeLastIterator<TSource>(IEnumerable<TSourc
             }
 
             Debug.Assert(queue.Count <= count);
-            do
-            {
-                yield return queue.Dequeue();
-            }
-            while (queue.Count > 0);
+
+            return queue;
```

Comment:
Thank you, but this change has a couple of problems:
1. It'll cause the input source to be enumerated as part of calling TakeLast, rather than as part of the first call to MoveNext on the eventual enumerator.
2. It'll result in the actual `Queue<T>` being returned, which different corner-case enumeration semantics than does the compiler-generated iterator, and which could be downcast back to `Queue<T>` and modified by the caller.

Reply:
1. Right
2. Concur